client/verta/verta/_pipeline_orchestrator/_orchestrator.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+
+import abc
+from typing import Any, Dict, Optional
+
+from verta._internal_utils import _utils
+from verta.external.cpython.graphlib import TopologicalSorter
+
+from ._step_handler import (
+    _StepHandlerBase,
+    ModelObjectStepHandler,
+)
+
+
+class _OrchestratorBase(abc.ABC):
```

Comment:
Basically, there's

- `DeployedOrchestrator` to orchestrate deployed model containers (stub for now)
- `LocalOrchestrator` to orchestrate locally-instantiated model containers

These subclasses implement logic to prepare step handlers. This base class (`_OrchestratorBase`) has the logic to prepare and run the pipeline.